onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -195,6 +195,7 @@ def optimize_model(
     num_heads: int = 0,
     hidden_size: int = 0,
     optimization_options: Optional[FusionOptions] = None,
+    optimized_model_path: Optional[str] = None,
```

Comment:
Note that the temp model will be removed (see line 300 below). I suggest that we do not change the interface. 
Instead, we can use a temp directory like:
```
   import tempfile
   with tempfile.TemporaryDirectory() as tmp_dir:
        # move the function code here.   
```

Reply:
I will revert the change to API signature. Actually, I prefer the tempfile solution. The only difference is that we could not use with clause since the model need be manually removed.